test/relationals.ll

```diff
@@ -14,17 +14,17 @@ declare dso_local spir_func <4 x i8> @_Z16__spirv_IsFiniteIDv4_aDv4_fET_T0_(<4 x
 declare dso_local spir_func <4 x i8> @_Z16__spirv_IsNormalIDv4_aDv4_fET_T0_(<4 x float>)
 declare dso_local spir_func <4 x i8> @_Z18__spirv_SignBitSetIDv4_aDv4_fET_T0_(<4 x float>)
 
-; CHECK-SPV-LLVM: call spir_func <4 x i32> @_Z13__spirv_IsNanDv4_f
-; CHECK-SPV-LLVM: call spir_func <4 x i32> @_Z13__spirv_IsInfDv4_f
-; CHECK-SPV-LLVM: call spir_func <4 x i32> @_Z16__spirv_IsFiniteDv4_f
-; CHECK-SPV-LLVM: call spir_func <4 x i32> @_Z16__spirv_IsNormalDv4_f
-; CHECK-SPV-LLVM: call spir_func <4 x i32> @_Z18__spirv_SignBitSetDv4_f
+; CHECK-SPV-LLVM: call spir_func <4 x i1> @_Z13__spirv_IsNanDv4_f
+; CHECK-SPV-LLVM: call spir_func <4 x i1> @_Z13__spirv_IsInfDv4_f
+; CHECK-SPV-LLVM: call spir_func <4 x i1> @_Z16__spirv_IsFiniteDv4_f
+; CHECK-SPV-LLVM: call spir_func <4 x i1> @_Z16__spirv_IsNormalDv4_f
+; CHECK-SPV-LLVM: call spir_func <4 x i1> @_Z18__spirv_SignBitSetDv4_f
```

Comment:
Sorry jumping into this late, is `i1` the type we want for builtins returning bools ?

We need to factor in the capability of a frontend to emit the correct return type. While `i1` and vectors of `i1` could be used, it is not a given the source language will allow this expression. The frontend can always be taught this, but that means you need to change the compiler.

AFAIK, that should be OK from a opencl C or C++ point of view, but the translator is not limited to these input languages.

@AnastasiaStulova @bader do you have any thoughts on that ?

Reply:
OpenCL frontend won't produce this IR because the functions return int type:
https://godbolt.org/z/bqz7T9cjv

Reply:
I agree with @Naghasan. @hewj03, could you clarify how do you see clang-based front-end to emit vector of i1? Anyway to store such values in memory i1 must be extended to i8 (at least).

Why it's only SPIR->OCL translation? We can argue that front-end issues are not relevant and these built-ins can be implemented in LLVM IR. But I don't think it's a good idea to put such restriction. In addition to that this change breaks code consistency after round-trip translation <n x i8> -> SPIR-V -> <n x i1>.

I suggest keeping existing return type for SPIR-V built-ins as update OpenCL built-ins only as described in the PR description.

Reply:
@Naghasan @bader thank you for catching this. This patch didn't intend to change spirv-friendly IR. Please review the new change.

i1 is from https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/5f289bae97f2815199a555a0c8151d7e2f36305c/lib/SPIRV/SPIRVReader.cpp#L375

Reply:
llvm based backend can consume spirv-friendly IR as input. If a builtin returning vector of i1 is not inlined, as I know, x86 backend won't has abi issue, but I don't know about other backends.

Reply:
> OpenCL frontend won't produce this IR because the functions return int type:
https://godbolt.org/z/bqz7T9cjv

@AnastasiaStulova your example is about the opencl c builtins. My point is about the SPIR-V insn https://www.khronos.org/registry/SPIR-V/specs/unified1/SPIRV.html#OpIsNan

Specifically:
```
Result Type must be a scalar or vector of Boolean type.
```

```
In addition to that this change breaks code consistency after round-trip translation

 I suggest keeping existing return type for SPIR-V built-ins
```

I don't think it makes sense for SPIR-V builtin functions to not follow the SPIR-V spec...

------

My point is what should a boolean maps to when it is taken as input or output of a SPIR-V builtin function, and more specifically: would mapping this to `i1` cause be a problem for a frontend to emit these ? 

Seems clang still don't allow for vector of bools https://godbolt.org/z/8Ts1c68Ta, so i1 is a problem in general.

Reply:
Not sure if there was an argument, but just a note:
I believe, that SPIR-V spec is not violated here see in :
```
; CHECK-SPIRV: {{[0-9]+}} TypeBool [[TBool:[0-9]+]]
; CHECK-SPIRV: {{[0-9]+}} IsNan [[TBoolVec]] [[IsNanRes:[0-9]+]]
```
so since the SPIR-V module is valid -the SPIR-V consumer can do whatever reasonable operations with it, for this case transforming a vector of i1 to a vector of i8.